src/erlcloud_s3.erl

```diff
@@ -1156,3 +1165,35 @@ aws_region_from_host(Host) ->
         _ ->
             "us-east-1"
     end.
+
+%%
+%% http://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html
+%% http://docs.aws.amazon.com/AmazonS3/latest/dev/Redirects.html
+%% Note: redirect response is handled only once.
```

Comment:
just reading around I think this comment is quite miss-leading since it redirect once but inside `s3_request4_no_update` there is a retry logic... which should be checked against 301-307 codes